code/game/turfs/space/transit.dm

```diff
@@ -59,7 +59,8 @@
 	AM.newtonian_move(dir)
 
 /turf/open/space/transit/CanBuildHere()
-	return istype(get_area(src), /area/shuttle)
+	var/area/current = get_area(src)
+	return istype(current, /area/shuttle) && !istype(current,/area/shuttle/transit)
```

Comment:
I redid the test by deconstructing the shuttle floor after launch. Reconstructing works then. I see your point.

But with this current PR you are unable to repair any breaches created before the shuttle launches, which is most damage to the shuttle occurs anyways and is not an ignorable case.